cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,172 @@
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "update_and_reboot.yaml")),
+            )
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster AMI",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+            )
+
+            components = [
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("UpdateAndReboot")
+                ),
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                ),
+            ]
+        else:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster custom AMI",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+            )
+
+            components = [
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                )
+            ]
+
+        # ImageRecipe
+        imagebuilder.CfnImageRecipe(
+            self,
+            id="PClusterImageRecipe",
+            name="PCluster-2-10-1" + str(hash(imagebuild)),
+            version="0.0.1",
+            parent_image=core.Fn.sub(
+                imagebuild.config.build_config.parent_image
+                or "arn:${AWS::Partition}:imagebuilder:${AWS::Region}:aws:image/amazon-linux-2-x86/x.x.x"
+            ),
+            components=components,
+            block_device_mappings=[
+                imagebuilder.CfnImageRecipe.InstanceBlockDeviceMappingProperty(
+                    device_name="/dev/xvda",
+                    ebs=ebs,
+                )
+            ],
+            tags={"ParallelClusterVersion": "2.10.1"},
+        )
+
+        # Image
+        imagebuilder.CfnImage(
+            self,
+            id="PClusterImage",
+            image_recipe_arn=core.Fn.ref("PClusterImageRecipe"),
+            infrastructure_configuration_arn=core.Fn.ref("PClusterImageInfrastructureConfiguration"),
+        )
+
+        # AWS Systems Manager
+        ssm.CfnParameter(
+            self,
+            id="PClusterParameter",
+            description="Image Id for PCluster",
+            name="/Test/Images/PCluster",
+            type="String",
+            value=core.Fn.get_att("PClusterImage", "ImageId").to_string(),
+        )
+
+
+def get_component_data(source_dir, file_name):
```

Comment:
minor: let's add an underscore prefix for private functions (functions used in this module only).

Reply:
Done